Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -162,6 +163,17 @@ - (void)resume
     [self.sessionReplay resume];
 }
 
+- (void)sentrySessionEnded:(SentrySession *)session {
+    [self stop];
+    [NSNotificationCenter.defaultCenter removeObserver:self];
```

Comment:
`m`: Please only use `removeObserver self` in `dealloc`. Instead, remove it with the most specific detail possible.

See: https://developer.apple.com/documentation/foundation/nsnotificationcenter/1413994-removeobserver#discussion
> When removing an observer, remove it with the most specific detail possible. For example, if you used a name and object to register the observer, use [removeObserver:name:object:](https://developer.apple.com/documentation/foundation/nsnotificationcenter/1407263-removeobserver) with the name and object.